src/System.Management.Automation/engine/Modules/GetModuleCommand.cs

```diff
@@ -489,50 +489,55 @@ private void GetLoadedModules(string[] names, IDictionary<string, ModuleSpecific
-        /// directory that do not explicitly support the current PowerShell edition.
-        /// </returns>
-        private IEnumerable<PSModuleInfo> FilterModulesForEdition(IEnumerable<PSModuleInfo> modules)
+        /// <param name="modules">The modules to filter incompatible examples from.</param>
+        /// <returns>All modules that are either not on the checked path or are compatible with the current PowerShell edition.</returns>
+        private IEnumerable<PSModuleInfo> FilterModulesForIncompatibleOnEditionCheckedPath(IEnumerable<PSModuleInfo> modules)
         {
-            // If no PSEdition is given, we only need to check the System32 path inclusions
-            if (String.IsNullOrEmpty(PSEdition))
+            Dbg.Assert(!SkipEditionCheck, $"Caller to verify that {nameof(SkipEditionCheck)} is false");
+
+            string currentPSEdition = PSVersionInfo.PSEdition;
+            foreach (PSModuleInfo module in modules)
             {
-                string currentPSEdition = PSVersionInfo.PSEdition;
-                foreach (PSModuleInfo module in modules)
+                IEnumerable<string> moduleCompatibleEditions = module.CompatiblePSEditions.Any() ? module.CompatiblePSEditions : DefaultCompatiblePSEditions;
+                if (moduleCompatibleEditions.Contains(currentPSEdition, StringComparer.OrdinalIgnoreCase))
                 {
-                    // If the module is not from a System32 path or if we have been told not to perform the check, it matches
-                    if (SkipEditionCheck || !module.IsLoadedFromCompatibilityCheckedPath)
-                    {
-                        yield return module;
-                        continue;
-                    }
-
-                    // Otherwise check modules on the System32 module path
-                    IEnumerable<string> moduleCompatibleEditions = module.CompatiblePSEditions.Any() ? module.CompatiblePSEditions : DefaultCompatiblePSEditions;
-                    if (moduleCompatibleEditions.Contains(currentPSEdition, StringComparer.OrdinalIgnoreCase))
-                    {
-                        yield return module;
-                    }
+                    yield return module;
                 }
-
-                yield break;
             }
+        }
+
+        /// <summary>
+        /// Filter an enumeration of modules based on what PowerShell editions they are compatible with.
+        /// If the PSEdition parameter has been given, filter modules based on that.
+        /// </summary>
+        /// <param name="modules">The modules to filter by edition.</param>
+        /// <returns> All modules meeting the PSEdition constraint.</returns>
+        private IEnumerable<PSModuleInfo> FilterModulesForEdition(IEnumerable<PSModuleInfo> modules)
+        {
+            Dbg.Assert(!String.IsNullOrEmpty(PSEdition), $"Caller to verify that {nameof(PSEdition)} is not null or empty");
```

Comment:
The same. What is benefits from moving the check on up level?

Reply:
The callee assertion seems to be a common occurrence in the codebase -- I could have included a comment, but thought an assertion was better as "zero-cost executable documentation".

Putting the check inside the method means needing to have a separate `for`-loop to handle the unchecked enumeration (since we can't use both `return` and `yield return` in the same method). Which doubles the size of the method and I think makes it less readable too. Doing it like this, we can avoid the whole method call ahead of time and keep the method itself cohesive.

Reply:
`return` -> `yield break`?

> we can avoid the whole method call 

I think a performance is not so important here.

Reply:
Well `yield break` will return an empty enumeration. But if `PSEdition` is not set you don't want that, you want everything. So you need another `for`-loop just for the case where it's not set, which seems pretty ugly to me compared to just opting out of the filter.

Yes I very much agree that the performance gain is minimal and not needed. Just pointing out a separate virtue of the approach I took.